src/language_features/completion.rs

```diff
@@ -196,21 +192,23 @@ pub fn editor_completion(
             // we'll need to perform some transformations on the completion commands.
             if ctx.config.snippet_support && x.insert_text_format == Some(InsertTextFormat::SNIPPET)
             {
+                lazy_static::lazy_static! {
+                    static ref SNIPPET_TABSTOP_RE: Regex = Regex::new(r"\$(?P<i>\d+)").unwrap();
+                    static ref SNIPPET_PLACEHOLDER_RE: Regex = Regex::new(r"\$\{(?P<i>\d+):?(?P<placeholder>[^}]+)\}").unwrap();
+                }
                 let snippet = insert_text;
-                let insert_text = snippet_prefix_re
-                    .find(&snippet)
-                    .map(|x| x.as_str())
-                    .unwrap_or(&snippet);
+                let insert_text = SNIPPET_TABSTOP_RE.replace_all(&snippet, "");
+                let insert_text = SNIPPET_PLACEHOLDER_RE.replace_all(&insert_text, "$placeholder");
+                // Theres some issue with multiline insert texts, and they also dont work well in the UI, so display on one line
+                let insert_text = insert_text.replace('\n', "");
```

Comment:
This might be a problem yes, but im not really sure how to solve it. We could trim it to just the first line, or even use ellipsis and set a max length.

We could also experiment with moving the cursor to the beginning, but i would prefer to avoid messing too much with the cursor and such while selecting completions.

It may also be possible to actually insert multiline text, but this is also not really a good experience in kakounes completion system, as the inserted text is hidden by the menu